airflow/models/trigger.py

```diff
@@ -71,27 +72,30 @@ def __init__(self, classpath: str, kwargs: dict[str, Any], created_date: datetim
         self.kwargs = kwargs
         self.created_date = created_date or timezone.utcnow()
 
-    @classmethod
-    def from_object(cls, trigger: BaseTrigger):
+    @staticmethod
```

Comment:
Just a nit:
`@internal_api_call` works fine with `@classmethod`
see
https://github.com/apache/airflow/blob/main/airflow/dag_processing/processor.py#L368
so these changes are not needed - I am not sure if it can break anything, unless someone decides to inherit from `Trigger` class

Reply:
That's a good nit. I must have missed it. Thanks for that! 
I reverted back to @classmethod. No need to take the risk :)